start.php

```diff
@@ -31,11 +31,36 @@
 
 	global $fs_active_plugins;
 
-	$this_sdk_relative_path = plugin_basename( dirname( __FILE__ ) );
+    $paths = array(

```

Comment:
I don't understand why:

> If I put the normalization code after line 63 as you mentioned, the paths will not be normalized in the theme's SDK's starter file if a plugin that uses an older SDK has already set the $fs_active_plugins global variable.

What if you take all the following code:

    $paths = array(
        'file_path'          => __FILE__,
        'current_theme_path' => trailingslashit( get_stylesheet_directory() ),
        'themes_directory'   => trailingslashit( get_theme_root() )
    );
    if ( function_exists( 'wp_normalize_path' ) ) {
        $paths = array_map( 'wp_normalize_path', $paths );
    } else {
        foreach ( $paths as $type => $path ) {
            $paths[ $type ] = str_replace( '\\', '/', $path );
            $paths[ $type ] = preg_replace( '|/+|', '/', $path );
        }
    }
    $fs_root_path = dirname( $paths[ 'file_path'] );
    if ( false !== strpos( $fs_root_path, $paths['current_theme_path'] ) ) {
	    $this_sdk_relative_path = '../themes/' . str_replace( $paths['themes_directory'], '', $fs_root_path );
        $is_theme = true;
    } else {
        $this_sdk_relative_path = plugin_basename( $fs_root_path );
        $is_theme = false;
    }

And move it below:

	if ( ! function_exists( 'fs_find_direct_caller_plugin_file' ) ) {
		require_once dirname( __FILE__ ) . '/includes/supplements/fs-essential-functions-1.1.7.1.php';
	}

Will it not work?